### PR TITLE
refactor: move isPrivateCluster to types.go

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**Reason for Change**:

Noticed that some isPrivateCluster assignments were missing nil checks and got carried away... This makes sure the function is in one central place and is unit tested.

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [x] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/api/types_test.go [hunk 1]
```diff
@@ -873,6 +873,65 @@ func TestRequireRouteTable(t *testing.T) {
...
 
+func TestIsPrivateCluster(t *testing.T) {
+	cases := []struct {
+		p        Properties
+		expected bool
+	}{
+		{
+			p: Properties{
+				OrchestratorProfile: &OrchestratorProfile{
+					OrchestratorType: DCOS,
```
Comment (jackfrancis) [inline pkg/api/types_test.go]: I'd kinda rather we're testing the `Kubernetes` default case (no `PrivateCluster` object in the api model at all).

**Comment by mboersma** [comment]
lgtm

**Pull request merged**